lib/fog/linode/requests/compute/avail_kernels.rb

```diff
@@ -5,27 +5,30 @@ class Real
         # Get available kernels
         #
         # ==== Parameters
-        # * kernelId<~Integer>: id to limit results to
+        # * options<~Hash>
+        #   * isXen<~Boolean> Show or hide Xen compatible kernels
+        #   * isKVM<~Boolean> Show or hide KVM compatible kernels
         #
         # ==== Returns
         # * response<~Excon::Response>:
         #   * body<~Array>:
         # TODO: docs
-        def avail_kernels(kernel_id=nil)
-          options = {}
-          if kernel_id
-            options.merge!(:kernelId => kernel_id)
-          end
+        def avail_kernels(options={})
+          # avail.kernels used to accept a kernelId parameter (now removed)
+          raise Fog::Errors::Error.new('avail_kernels no longer accepts a kernelId parameter') unless !options || options.is_a?(Hash)
```

Comment:
@plribeiro3000 It is possible that any existing uses of `avail_kernels` could be using types other than `Integer` for the `kernelId` (Excon calls `to_s` on parameter values). I therefore prefer to reject anything that is not a `Hash` or `nil`.